src/NuGet.Core/NuGet.Resolver/ResolverUtility.cs

```diff
@@ -303,7 +303,7 @@ public static IEnumerable<ResolverPackage> TopologicalSort(IEnumerable<ResolverP
 
             // add any unsorted nodes onto the end of the result
             var sorted = new HashSet<string>(result.Select(n => n.Id), StringComparer.OrdinalIgnoreCase);
-            result.AddRange(nodes.Where(n => !sorted.Contains(n.Id)));
+            result.AddRange(nodes.Where(n => !sorted.Contains(n.Id)).OrderBy(p => p.Id, StringComparer.OrdinalIgnoreCase));
```

Comment:
nit: rename sorted - because it is not really sorted it just a hashset, which makes the orderby call confusing.

Also move the nodes.... call outside the addrange to a variable - so it is easy to inspect and debug (and you can call that sorted, because now it is)